docs/preview/features/secret-store/named-secret-providers.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "Named secret providers"
+layout: default
+---
+
+# Retrieve a specific secret provider from the secret store
+
+The default workings of the secret store, is that a set of secret providers are registered and the consumer gets access to all of secrets provided by `ISecretProvider`s.
+In some cases, you may want to retrieve a specific secret provider from the store. This can be because that secret provider has some functionality that the other providers doesn't have.
+
+In those cases, you can register your secret provider with a unique name. This name will be matched with the secret provider so that in a later stage, you can retrieve back your named provider.
+
+## Register a named secret provider
+
+Lets consider that you want explicitly use the built-in environment variables secret provider.
+First, the secret provider has to be registered with a unique name. 
+
+```csharp
+.ConfigureSecretStore((config, stores) =>
+{
+    stores.AddEnvironmentVariables(..., name: "environment-variables");
+})
+```
+
+## Retrieve the named secret provider
+
+Now that the named environment variables secret provider is registered, we should be able to retrieve this provider somewhere else in the application.
+Normally, you inject the `ISecretProvider` in your application service and accesses the secrets, but now you'll have to inject the `ISecretStore` interface to access the specific secret store operations.
```

Comment:
```suggestion

Instead of injecting `ISecretProvider` in your application to access secrets, we'll inject `ISecretStore` interface to retrieve named secret providers.
```